primitives/beefy/src/mmr.rs

```diff
@@ -26,14 +26,89 @@
 //! but we imagine they will be useful for other chains that either want to bridge with Polkadot
 //! or are completely standalone, but heavily inspired by Polkadot.
 
-use crate::{crypto::AuthorityId, ConsensusLog, MmrRootHash, Vec, BEEFY_ENGINE_ID};
-use codec::{Decode, Encode, MaxEncodedLen};
-use scale_info::TypeInfo;
+use crate::{ecdsa_crypto::AuthorityId, ConsensusLog, MmrRootHash, Vec, BEEFY_ENGINE_ID};
+use codec::{Decode, Encode, Error as CodecError, Input, MaxEncodedLen, Output};
+use scale_info::{build::Fields, Path, Type, TypeInfo};
 use sp_runtime::{
 	generic::OpaqueDigestItemId,
 	traits::{Block, Header},
 };
 
+use apk_proofs::KeysetCommitment;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+#[cfg(feature = "std")]
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+/// Wrapper around keyset commitment to provide scale codec for the commitment
+//#[cfg_attr(feature = "std", derive(Deserialize))]
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct AuthoritySetCommitment(pub KeysetCommitment);
+
+impl Encode for AuthoritySetCommitment {
```

Comment:
I did haphazard job implementing Scale for the APK Commitment so I can compile the code. I'm sure someone else on the team knows better how to implement Scale and Serialization related stuff, so please have a go at it.

Reply:
Related to this, @swasilyev is reluctant to add Scale to KeysetCommitment as it is substrate specific. Plus we should have Scale for Arkworks' CanonicalSerialize as we are scaling arkworks object other place (like new crypto host function for pairing etc).